account_invoice_margin/models/account_invoice.py

```diff
@@ -0,0 +1,121 @@
+# Copyright 2017 Sergio Teruel <sergio.teruel@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    margin = fields.Monetary(
+        compute="_compute_margin",
+        store=True,
+        currency_field="currency_id",
+    )
+
+    margin_signed = fields.Monetary(
+        compute="_compute_margin",
+        store=True,
+        currency_field="currency_id",
+    )
+
+    margin_percent = fields.Float(
+        string="Margin (%)",
+        digits="Product Price",
+        compute="_compute_margin",
+        store=True,
+    )
+
+    def _get_margin_applicable_lines(self):
+        self.ensure_one()
+        return self.invoice_line_ids
+
+    @api.depends(
+        "invoice_line_ids.margin",
+        "invoice_line_ids.margin_signed",
+        "invoice_line_ids.price_subtotal",
+    )
+    def _compute_margin(self):
+        for invoice in self:
+            margin = 0.0
+            margin_signed = 0.0
+            price_subtotal = 0.0
+            for line in invoice._get_margin_applicable_lines():
```

Comment:
```suggestion
            for line in invoice.line_ids.filtered(lambda x: x.is_margin_applicable):
```

What about to have a compute field on account.invoice.line. I think that it's a caracteristic of the line.

Just a proposal, not a blocking point.

Reply:
I'll take a look, thank you @legalsylvain for pointing it out.

Reply:
Well, in fact, the ideal should be to have a uniq function _is_applicable_margin / _get_margin_applicable_lines or whatever that could be called in the account.move margin module / sale.order margin module / pos.order margin module.

But I'm not sure how to proceed.